src/main/java/org/trigon/hardware/misc/leds/CANdleLEDStrip.java

```diff
@@ -0,0 +1,219 @@
+package org.trigon.hardware.misc.leds;
+
+import com.ctre.phoenix.led.*;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.util.Color;
+import org.trigon.hardware.RobotHardwareStats;
+
+import java.util.function.Supplier;
+
+/**
+ * A LED strip that is controlled by a CANdle, and uses AddressableLED for simulation.
+ */
+public class CANdleLEDStrip extends LEDStrip {
+    private static CANdle CANDLE;
+    private static int LAST_CREATED_LED_STRIP_ANIMATION_SLOT = 0;
+    private final int animationSlot;
+    private final AddressableLEDStrip simulationLEDStrip;
+
+    /**
+     * Sets the CANdle instance to be used for controlling the LED strips. Must be set before using any LED strips.
+     *
+     * @param candle the CANdle instance to be used
+     */
+    public static void setCandle(CANdle candle) {
+        CANDLE = candle;
+    }
+
+    /**
+     * Sets the simulation AddressableLED instance to be used for testing in simulation. Must be set before using any LED strips in simulation.
+     *
+     * @param simulationLEDStrip the AddressableLED instance to be used in simulation
+     */
+    public static void setSimulationLED(AddressableLED simulationLEDStrip) {
+        AddressableLEDStrip.setAddressableLED(simulationLEDStrip);
+    }
+
+    /**
+     * Sets the simulation AddressableLEDBuffer instance to be used for testing in simulation. Must be set before using any LED strips in simulation.
+     *
+     * @param simulationLEDBuffer the AddressableLED buffer instance to be used in simulation
+     */
+    public static void setSimulationLEDBuffer(AddressableLEDBuffer simulationLEDBuffer) {
+        AddressableLEDStrip.setAddressableLEDBuffer(simulationLEDBuffer);
+    }
```

Comment:
I feel like this can be automated in some way in the setCANdle method

Reply:
I tried it, but the problem is that the LEDBuffer needs the total amount of LEDs(all the strips) in it's constructer

Reply:
Then make one function that creates everything using only that number